pkg/kubectl/cmd/rollingupdate.go

```diff
@@ -161,7 +171,7 @@ func RunRollingUpdate(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, arg
 
 		if len(args) >= 2 {
 			newName = args[1]
-		} else {
+		} else if oldRc != nil {
```

Comment:
Might be time to break this function up. Too hard to follow the logic if you have to scroll thru multiple screen lengths.